Staking/contracts/StakingContract.sol

```diff
@@ -127,5 +132,30 @@ contract StakingContract {
         uint elapsedTime = block.timestamp - stakers[staker].startTime;
         return (stakers[staker].stakedAmount * interestRate * elapsedTime) / (365 days * baseUnit * 100);
     }
-}
 
+    // Add Reward function adds an additional reward to the contract balance which is divided
+    // proportionally among the stakers.
+    function addReward(uint256 rewardAmount) external onlyOwner {
+        require(rewardAmount > 0, "Invalid amount: Must add a positive amount");
+
+        // Transfer ERC20 tokens to the contract
+        require(rewardToken.transferFrom(msg.sender, address(this), rewardAmount), "Failed to transfer ERC20 tokens");
+
+        // Calculate total staked amount
+        uint totalStakedAmount = 0;
+        for (uint i = 0; i < addressStakers.length; i++) {
```

Comment:
We should avoid loops in contracts. Try to find a way to implement it without loop